app/src/main/java/com/schibsted/mappicker/MainActivity.kt

```diff
@@ -0,0 +1,135 @@
+import com.schibstedspain.leku.tracker.TrackEvents
+import java.util.UUID
+
+private const val MAP_BUTTON_REQUEST_CODE = 1
+private const val MAP_POIS_BUTTON_REQUEST_CODE = 2
+
+class MainActivity : AppCompatActivity() {
+
+    private val lekuPois: List<LekuPoi>
+        get() {
+            val pois = ArrayList<LekuPoi>()
+
+            val locationPoi1 = Location("leku")
+            locationPoi1.latitude = 41.4036339
+            locationPoi1.longitude = 2.1721618
+            val poi1 = LekuPoi(UUID.randomUUID().toString(), "Los bellota", locationPoi1)
+            pois.add(poi1)
+
+            val locationPoi2 = Location("leku")
+            locationPoi2.latitude = 41.4023265
+            locationPoi2.longitude = 2.1741417
+            val poi2 = LekuPoi(UUID.randomUUID().toString(), "Starbucks", locationPoi2)
+            poi2.address = "Plaça de la Sagrada Família, 19, 08013 Barcelona"
+            pois.add(poi2)
+
+            return pois
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        StrictMode.setThreadPolicy(
+                StrictMode.ThreadPolicy.Builder().detectAll().penaltyLog().build())
+        StrictMode.setVmPolicy(StrictMode.VmPolicy.Builder().detectAll().penaltyLog().build())
+        setContentView(R.layout.activity_main)
+        val mapButton = findViewById<View>(R.id.map_button)
+        mapButton.setOnClickListener { _ ->
+            val locationPickerIntent = LocationPickerActivity.Builder()
+                    .withLocation(41.4036299, 2.1743558)
+                    //.withGeolocApiKey("<PUT API KEY HERE>")
+                    .withSearchZone("es_ES")
+                    //.shouldReturnOkOnBackPressed()
+                    //.withStreetHidden()
+                    //.withCityHidden()
+                    //.withZipCodeHidden()
+                    //.withSatelliteViewHidden()
+                    //.withGooglePlacesEnabled()
+                    .build(applicationContext)
+
+            //this is optional if you want to return RESULT_OK if you don't set the latitude/longitude and click back button
+            locationPickerIntent.putExtra("test", "this is a test")
+
+            startActivityForResult(locationPickerIntent, MAP_BUTTON_REQUEST_CODE)
+        }
+
+        val mapPoisButton = findViewById<View>(R.id.map_button_with_pois)
+        mapPoisButton.setOnClickListener { _ ->
+            val locationPickerIntent = LocationPickerActivity.Builder()
+                    .withLocation(41.4036299, 2.1743558)
+                    .withPois(lekuPois)
+                    .build(applicationContext)
+
+            startActivityForResult(locationPickerIntent, MAP_POIS_BUTTON_REQUEST_CODE)
+        }
+
+        initializeLocationPickerTracker()
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
+        if (resultCode == Activity.RESULT_OK) {
+            Log.d("RESULT****", "OK")
```

Comment:
Shouldn't we remove this Logs? This logs in production no?

Reply:
This is only for the sample

Reply:
ok